### PR TITLE
Made a bunch of library code a bit safer

### PR DESCRIPTION
**Pull request by ryanpbrewster**
We shouldn't do things with inactive libraries. Somehow these ids are leaking
out, and there is code that is trying to use them. I made most of this code
more robust: where possible, it assumes that some of the library ids it has
could be invalid, and it does a `libraryIds.flatMap(libraryRepo.getActive)`
thing that drops the inactive ids.

@andrewconner @leogrim @camhashemi

### CODE REVIEW

**Comment by andrewconner** [comment]
:fireworks: :checkered_flag: :heart:

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminLibraryController.scala [hunk 1]
```diff
@@ -137,11 +137,11 @@ class AdminLibraryController @Inject() (
         val allLibraryIds = pagedLibrariesById.keySet ++ topDailyFollower.map(_._1) ++ topDailyKeeps.map(_._1) ++ hotToday.map(_._1)
         val librariesById = {
           val missingLibraryIds = allLibraryIds -- pagedLibrariesById.keys
-          val missingLibraries = if (missingLibraryIds.nonEmpty) libraryRepo.getLibraries(missingLibraryIds) else Map.empty
+          val missingLibraries = if (missingLibraryIds.nonEmpty) libraryRepo.getActiveByIds(missingLibraryIds) else Map.empty
           pagedLibrariesById ++ missingLibraries
         }
         val usersById = userRepo.getAllUsers(librariesById.values.map(_.ownerId).toSeq)
-        librariesById.mapValues(lib => buildLibStatistic(lib, usersById(lib.ownerId)))
+        librariesById.map { case (libId, lib) => libId -> buildLibStatistic(lib, usersById(lib.ownerId)) }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminLibraryController.scala:144]: `buildLibStatistic` uses an implicit session, and `mapValues` provides a lazy view. Those two things spell trouble.

I'm tempted to write an implicit extension that provides a `strictMapValues(f)` that wraps `.map { case (id, v) => id -> f(v) }`

**Comment by camhashemi** [comment]
thanks for this :+1:

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepDecorator.scala [hunk 2]
```diff
@@ -84,10 +84,11 @@ class KeepDecoratorImpl @Inject() (
           val keepers = keeps.map(_.userId).toSet // is this needed? need to double check, it may be redundant
           db.readOnlyMaster { implicit s => basicUserRepo.loadAll(keepersShown ++ libraryContributorsShown ++ libraryOwners ++ keepers) } //cached
         }
-        val idToBasicLibrary = idToLibrary.mapValues { library =>
-          val orgOpt = basicOrgByLibId.get(library.id.get)
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepDecorator.scala:88]: Same story, `mapValues` does not partner well with db sessions.

**Pull request merged**